source/extensions/filters/http/lua/config.cc

```diff
@@ -25,7 +24,7 @@ Router::RouteSpecificFilterConfigConstSharedPtr
 LuaFilterConfig::createRouteSpecificFilterConfigTyped(
     const envoy::extensions::filters::http::lua::v3::LuaPerRoute& proto_config,
     Server::Configuration::ServerFactoryContext& context, ProtobufMessage::ValidationVisitor&) {
-  return std::make_shared<FilterConfigPerRoute>(proto_config, context.threadLocal(), context.api());
+  return std::make_shared<FilterConfigPerRoute>(proto_config, context);
```

Comment:
Sorry I completely forgot that we do pre-create all of the per-route filter config on route table load, so yeah we don't need any new callbacks here I don't think.